service/r2/service.go

```diff
@@ -96,6 +99,39 @@ func newServiceMetrics(scope tally.Scope, samplingRate float64) serviceMetrics {
 	}
 }
 
+type route struct {
+	path   string
+	method string
+}
+
+var authorizationRegistry = map[route]auth.AuthorizationType{
+	// This validation route should only require read access.
+	{path: validateRuleSetPath, method: http.MethodPost}: auth.AuthorizationTypeReadOnly,
+}
+
+func defaultAuthorizationTypeForHTTPMethod(method string) auth.AuthorizationType {
```

Comment:
I'd still return an error here just to make it very explicit that if we don't see one of the expected methods, it should be an error, and check error in `registerRoute`.